ports/vcpkg-make/vcpkg_make_common.cmake

```
@@ -0,0 +1,660 @@
+        endif()
+
+    else()
+        # OSX dosn't like CMAKE_C(XX)_COMPILER (cc) in CC/CXX and rather wants to have gcc/g++
+        vcpkg_list(SET z_vcm_all_tools)
+        function(z_vcpkg_make_set_env envvar cmakevar)
+            set(prog "${VCPKG_DETECTED_CMAKE_${cmakevar}}")
+            if(NOT DEFINED ENV{${envvar}} AND NOT prog STREQUAL "")
+                vcpkg_list(APPEND z_vcm_all_tools "${prog}")
+                if(ARGN)
+                    string(APPEND prog " ${ARGN}")
+                endif()
+                set(z_vcm_all_tools "${z_vcm_all_tools}" PARENT_SCOPE)
+            endif()
+            z_vcpkg_append_to_configure_environment(configure_env "${envvar}" "${prog}")
+            set(configure_env "${configure_env}" PARENT_SCOPE)
+        endfunction()
+
+        z_vcpkg_make_set_env(CC C_COMPILER ${ABIFLAGS_${arg_CONFIG}})
+        z_vcpkg_make_set_env(CXX CXX_COMPILER ${ABIFLAGS_${arg_CONFIG}})
+        if(NOT arg_BUILD_TRIPLET MATCHES "--host")
+            z_vcpkg_make_set_env(CC_FOR_BUILD C_COMPILER ${ABIFLAGS_${arg_CONFIG}})
+            z_vcpkg_make_set_env(CPP_FOR_BUILD C_COMPILER "-E" ${ABIFLAGS_${arg_CONFIG}})
+            z_vcpkg_make_set_env(CXX_FOR_BUILD CXX_COMPILER ${ABIFLAGS_${arg_CONFIG}})
+        else()
+            set(ENV{CC_FOR_BUILD} "umask 0 | touch a.out | touch conftest${VCPKG_HOST_EXECUTABLE_SUFFIX} | true")
+            set(ENV{CPP_FOR_BUILD} "umask 0 | touch a.out | touch conftest${VCPKG_HOST_EXECUTABLE_SUFFIX} | true")
+            set(ENV{CXX_FOR_BUILD} "umask 0 | touch a.out | touch conftest${VCPKG_HOST_EXECUTABLE_SUFFIX} | true")
+        endif()
+        if("ASM" IN_LIST arg_LANGUAGES)
+            z_vcpkg_make_set_env(CCAS ASM_COMPILER "-c" ${ABIFLAGS_${arg_CONFIG}})
+            z_vcpkg_make_set_env(AS ASM_COMPILER "-c" ${ABIFLAGS_${arg_CONFIG}})
+        endif()
+        z_vcpkg_make_set_env(NM NM)
+        z_vcpkg_make_set_env(RC RC)
+        z_vcpkg_make_set_env(WINDRES RC)
+        z_vcpkg_make_set_env(DLLTOOL DLLTOOL)
+        z_vcpkg_make_set_env(STRIP STRIP)
+        z_vcpkg_make_set_env(OBJDUMP OBJDUMP)
+        z_vcpkg_make_set_env(RANLIB RANLIB)
+        z_vcpkg_make_set_env(AR AR)
+        z_vcpkg_make_set_env(LD LINKER)
+        unset(z_vcpkg_make_set_env)
+
+        list(FILTER z_vcm_all_tools INCLUDE REGEX " ")
+        if(z_vcm_all_tools)
+            list(REMOVE_DUPLICATES z_vcm_all_tools)
+            list(JOIN z_vcm_all_tools "\n   " tools)
+            message(STATUS "Warning: Tools with embedded space may be handled incorrectly by configure:\n   ${tools}")
+        endif()
+    endif()
+    list(JOIN configure_env " " configure_env)
+    set("${out_env}" "${configure_env}" PARENT_SCOPE)
+endfunction()
+
+function(z_vcpkg_make_prepare_link_flags)
+    cmake_parse_arguments(PARSE_ARGV 0 ARG
+        ""
+        "IN_OUT_VAR;X_VCPKG_TRANSFORM_LIBS;VCPKG_TARGET_IS_WINDOWS;VCPKG_TARGET_IS_MINGW;VCPKG_LIBRARY_LINKAGE"

```

Confusing...
```suggestion
        "IN_OUT_VAR;TRANSFORM_LIBS;TARGET_IS_WINDOWS;TARGET_IS_MINGW;LIBRARY_LINKAGE"
```
... or maybe not use args at all. Calling sites, in particual tests, might override vars within `block()` scopes.

---

Can you explain?

---

`X_VCPKG_TRANSFORM_LIBS` has a prefix for no good reason, but this is in my other comment.

`VCPKG_TARGET_IS_WINDOWS;VCPKG_TARGET_IS_MINGW;VCPKG_LIBRARY_LINKAGE` have names which are very similar to global variables, inviting accidental using the wrong identifier. I don't see why they should be passed at all in regular code. I see a benefit for test code, but tests could use
~~~cmake
block(SCOPE_FOR variables)
    set(VCPKG_TARGET_IS_MINGW 1) # but only after stopping `"STREQUAL "ON"` 
    z_vcpkg_make_prepare_link_flags(...)
endblock()
~~~